### PR TITLE
[1-setup] Set up a gradle project with three services and common module

### PR DESCRIPTION
**Pull request by a-davydenko**

### CODE REVIEW

File: settings.gradle [hunk 1]
```diff
@@ -1 +1,6 @@
 rootProject.name = 'tic-tac-toe'
+
+include 'battle-service'
+include 'statistics-service'
+include 'user-service'
+include 'common'
```
Comment (Volad) [inline settings.gradle]: let's probably remove common module. we don't know so far if we will need this at all.
also "common" is a very wide term ,which actually doesn't help to solve specific problem, but rise fault wished to put as much as possible there. which usually leads to more problems.

File: statistics-service/build.gradle [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        extendsFrom annotationProcessor
+    }
+}
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    implementation project(':common')
```
Comment (Volad) [inline statistics-service/build.gradle]: in multi module gradle project you may have a root build.gradle which will hold all common dependencies for all sub-modules.


```
-./build.gradle
-./settings.gradle
-./battle-service
   |--/build.gradle
-./user-service
   |--/build.gradle

```

so in root build.gradle you may put common dependencies, ie lombok,spring,springboot etc.

File: user-service/src/main/java/com/adavydenko/userservice/UserServiceApplication.java [hunk 3]
```diff
@@ -1,22 +1,23 @@
-package com.adavydenko.tictactoe;
```
Comment (Volad) [inline user-service/src/main/java/com/adavydenko/userservice/UserServiceApplication.java]: i would keep tictactoe as part of the package.. you may also make it as part of your artifact group in common build.gradle file.

File: battle-service/build.gradle [hunk 4]
```diff
@@ -19,12 +19,16 @@ repositories {
...
 dependencies {
+    implementation project(':common')
+    implementation 'org.springframework.boot:spring-boot-starter'
     implementation 'org.springframework.boot:spring-boot-starter-web'
     compileOnly 'org.projectlombok:lombok'
     annotationProcessor 'org.projectlombok:lombok'
     testImplementation 'org.springframework.boot:spring-boot-starter-test'
 }
 
+tasks.register("prepareKotlinBuildScriptModel") {}
```
Comment (Volad) [inline battle-service/build.gradle]: why you need this?

File: user-service/build.gradle [hunk 5]
```diff
@@ -0,0 +1,34 @@
+plugins {
```
Comment (Volad) [inline user-service/build.gradle:1]: most of this could be part of globabl build.gradle and shared between all subprojects

Comment (Volad) [reply user-service/build.gradle:1]: i meant this one:

[Cross Configuration](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#sec:convention_plugins_vs_cross_configuration
)

You may want to try  more "smart way"
[Convention Plugin](https://docs.gradle.org/current/userguide/sharing_build_logic_between_subprojects.html#sec:convention_plugins)
[sample](
https://docs.gradle.org/current/samples/sample_convention_plugins.html)

However i never did option #2, due to subprojects were well enough for my needs.

File: battle-service/build.gradle (hunk 4 above)
Comment (a-davydenko) [reply battle-service/build.gradle:30]: i had an error saying "Task 'prepareKotlinBuildScriptModel' not found in project ':battle-service'"

Comment (a-davydenko) [reply battle-service/build.gradle]: > why you need this?

i had an error saying "Task 'prepareKotlinBuildScriptModel' not found in project ':battle-service'"

File: battle-service/build.gradle [hunk 6]
```diff
@@ -0,0 +1,4 @@
+plugins {
+    id 'org.springframework.boot' version '3.0.5'
```
Comment (Volad) [inline battle-service/build.gradle:2]: you can move this to root build gradle

**Review by Volad — APPROVED** [review]

**Pull request merged**